pkg/services/ngalert/state/cache.go

```diff
@@ -15,15 +15,15 @@ import (
 )
 
 type cache struct {
-	states    map[string]*State
+	states    map[int64]map[string]map[string]*State
```

Comment:
Okay hash is starting to look a lot better :-) Might help for now to add comment here from your PR prose: `orgID>alertRuleUID>stateId`